app/lib/bk/compat/parsers/bitrise.rb

```diff
@@ -39,8 +39,11 @@ def parse
         workflows = @config.fetch('workflows', {})
         bk_steps = workflows.map { |wf_name, wf_config| parse_workflow(wf_name, wf_config) }
 
+        stack_value = @config.dig('meta', 'bitrise.io', 'stack')
+        agents = stack_value ? { stack: stack_value } : {}
         Pipeline.new(
-          steps: bk_steps
+          steps: bk_steps,
+          agents: agents || {}
```

Comment:
you are already ensuring that `agents` is a dictionary in line 43 so this is not necessary:

```suggestion
          agents: agents
```